pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -63,34 +93,73 @@ let
   # Java toolchain used for the build and tests
   javaToolchain = "@bazel_tools//tools/jdk:toolchain_host${buildJdkName}";
 
+  platforms = lib.platforms.linux ++ lib.platforms.darwin;
+
 in
 stdenv.mkDerivation rec {
 
-  version = "0.24.0";
+  version = "0.26.0";
 
   meta = with lib; {
     homepage = "https://github.com/bazelbuild/bazel/";
     description = "Build tool that builds code quickly and reliably";
     license = licenses.asl20;
     maintainers = [ maintainers.mboes ];
-    platforms = platforms.linux ++ platforms.darwin;
+    inherit platforms;
   };
 
   # Additional tests that check bazel’s functionality. Execute
   #
   #     nix-build . -A bazel.tests
   #
   # in the nixpkgs checkout root to exercise them locally.
-  passthru.tests = {
-    pythonBinPath = callPackage ./python-bin-path-test.nix {};
-    bashTools = callPackage ./bash-tools-test.nix {};
-  };
+  passthru.tests =
+    let
+    runLocal = name: script: runCommandCC name {
+      preferLocalBuild = true;
+      meta.platforms = platforms;
+    } script;
+    # bazel wants to extract itself into $HOME every time it runs,
+    # so let’s do that only once.
+    homeDir = runLocal "bazel-extracted-homedir" ''
+      export HOME=$out
+      ${bazel}/bin/bazel version
+    '';
+    bazelTest = { name, bazelScript, workspaceDir }:
+      runLocal name ''
+        # set up home with pre-unpacked bazel
+        export HOME=$(mktemp -d)
+        # yes, this path is kinda magic. Sorry.
+        install_dir=.cache/bazel/_bazel_nixbld
+        mkdir -p $HOME/$install_dir
+        cp -R ${homeDir}/$install_dir/install $HOME/$install_dir
```

Comment:
the `homeDir` itself already layed out as `$HOME/.cache/...`, why not just `cp -R ${homeDir} $HOME` (obviously rm $HOME before you do)?

Reply:
Ah, sorry, I missed the review and force-pushed a few changes with rebases. I had to abstract this out anyway, because bazel *of course* does not use `$HOME` for its extraction on MacOS, but `/var/tmp` …

Reply:
:man_facepalming: 

Thank you for working on this @Profpatsch!!